### PR TITLE
Convert `cms-agilitycms` example to TypeScript

### PR DESCRIPTION
**Pull request by maxproske**
Converted `cms-agilitycms` example to TypeScript to match Contribution docs.

## Documentation / Examples

- [X] Make sure the linting passes by running `pnpm lint`
- [X] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: examples/cms-agilitycms/environment.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+declare namespace NodeJS {
+  export interface ProcessEnv {
+    readonly NODE_ENV: 'development' | 'production' | 'test'
```
Comment (balazsorban44) [inline examples/cms-agilitycms/environment.d.ts:3]: This should already be present.
```suggestion
```

**Review by balazsorban44 — DISMISSED** [review]
Thanks! Minor comment, but looks good.

**Comment by maxproske** [comment]
@balazsorban44 Good call. Fixed!

**Review by balazsorban44 — DISMISSED** [review]

File: examples/cms-agilitycms/pages/api/preview.ts [hunk 2]
```diff
@@ -19,6 +23,6 @@ export default async function handler(req, res) {
   res.setPreviewData({})
 
   // Redirect to the slug
-  res.writeHead(307, { Location: validationResp.slug })
+  res.writeHead(307, { Location: req.query.slug })
```
Comment (ijjk) [inline examples/cms-agilitycms/pages/api/preview.ts:26]:
```suggestion
  if (!('slug' in validationResp)) {
    throw new Error('invariant missing slug in validation response')
  }
  res.writeHead(307, { Location: validationResp.slug })
```
I think we still want to use `validationResp.slug` here as it helps stress we should avoid redirecting to un-sanitized user input.

Comment (maxproske) [reply examples/cms-agilitycms/pages/api/preview.ts]: Thanks! I felt slimy trusting un-sanitized user input, so good to know about `validationResp.slug`.

**Review by ijjk — APPROVED** [review]

**Review by ijjk — APPROVED** [review]
Thanks!

**Pull request merged**